src/users/__tests__/userHelpers-test.js

```diff
@@ -180,24 +217,78 @@ describe('sortUserList', () => {
       { full_name: 'Bob', email: 'bob@example.com' },
       { full_name: 'Rick', email: 'rick@example.com' },
     ]);
+
     const presences = {
-      'mark@example.com': { aggregated: { status: 'offline' } },
+      'mark@example.com': {
+        aggregated: { status: 'offline', client: 'ZulipMobile', timestamp: Date.now() },
+        ZulipMobile: {
+          status: 'offline',
+          client: 'ZulipMobile',
+          timestamp: Date.now(),
+          pushable: false,
+        },
+      },
       'john@example.com': {
-        aggregated: { status: 'active', timestamp: Date.now() / 1000 - 120 * 60 },
+        aggregated: {
+          status: 'active',
+          client: 'website',
+          timestamp: Date.now() / 1000 - 120 * 60,
+        },
+        website: {
+          status: 'active',
+          client: 'website',
+          timestamp: Date.now() / 1000 - 120 * 60,
+          pushable: false,
+        },
+      },
+      'bob@example.com': {
+        aggregated: {
+          status: 'idle',
+          client: 'ZulipMobile',
+          timestamp: Date.now() / 1000 - 20 * 60,
+        },
+        ZulipMobile: {
+          status: 'idle',
+          client: 'ZulipMobile',
+          timestamp: Date.now() / 1000 - 20 * 60,
+          pushable: true,
+        },
+      },
+      'rick@example.com': {
+        aggregated: { status: 'active', client: 'ZulipMobile', timestamp: Date.now() / 1000 },
+        ZulipMobile: {
+          status: 'active',
+          client: 'ZulipMobile',
+          timestamp: Date.now() / 1000,
+          pushable: true,
+        },
       },
-      'bob@example.com': { aggregated: { status: 'idle', timestamp: Date.now() / 1000 - 20 * 60 } },
-      'rick@example.com': { aggregated: { status: 'active', timestamp: Date.now() / 1000 } },
     };
+
     const shouldMatch = [
       { full_name: 'Rick', email: 'rick@example.com' },
       { full_name: 'Bob', email: 'bob@example.com' },
       { full_name: 'John', email: 'john@example.com' },
       { full_name: 'Mark', email: 'mark@example.com' },
     ];
 
-    const sortedUsers = sortUserList(users, presences);
-
-    expect(sortedUsers).toEqual(shouldMatch);
+    // PresenceState type is an exact object with indexers [email: string]: string, [client:string]:string
+    // so key should have a type of string. So email and presenceKey are defined with type string and then
+    // passed as a key: value to the respective objects.
+    const modifiedPresences = {};
+    Object.keys(presences).forEach(user => {
+      const userStatus = {};
+      const email: string = user;
+      const userPresence = presences[email];
+      Object.keys(userPresence).forEach(key => {
+        const presenceKey: string = key;
+        userStatus[presenceKey] = userPresence[presenceKey];
+      });
+      modifiedPresences[email] = userStatus;
+    });
```

Comment:
These steps are required for converting to exact Objects with specific indexers.